source/PlanetLabel.cpp

```diff
@@ -39,6 +40,41 @@ namespace {
 	const double LINE_GAP = 1.7;
 	const double GAP = 6.;
 	const double MIN_DISTANCE = 30.;
+
+	// Check if the given label for the given stellar object and direction overlaps with any other stellar object in the system.
+	bool Overlaps(const System &system, const StellarObject &object, double zoom, double width, int direction)
+	{
+		bool overlaps = false;
+
+		Point start = zoom * (object.Position() +
+			(object.Radius() + INNER_SPACE + LINE_GAP + LINE_LENGTH) * Angle(LINE_ANGLE[direction]).Unit());
+		Point unit(LINE_ANGLE[direction] > 180. ? -1. : 1., 0.);
+		Point end = start + unit * width;
+
+		for(const StellarObject &other : system.Objects())
+		{
+			if(&other == &object)
+				continue;
+
+			double minDistance = (other.Radius() + MIN_DISTANCE) * zoom;
+
+			Point otherPos = other.Position() * zoom;
+			double startDistance = otherPos.Distance(start);
+			double endDistance = otherPos.Distance(end);
+			overlaps |= (startDistance < minDistance || endDistance < minDistance);
+			double projection = (otherPos - start).Dot(unit);
+
+			if(projection > 0. && projection < width)
```

Comment:
This code appears to check for "overlap with the middle of the label, when the end and/or start might not overlap". Maybe good to add a comment above this code to describe this for the maintainers that need to deal with this code in the future.